sw/device/lib/testing/test_framework/BUILD

```diff
@@ -44,3 +44,79 @@ cc_library(
         "//sw/device/lib/runtime:print",
     ],
 )
+
+# TODO(lowRISC/opentitan#7972): clean up check.h after completing the
+# migration to bazel.
+cc_library(
+    name = "check",
+    hdrs = [
+        "test_status.h",
+        "//sw/device/lib/testing:check",
+    ],
+    visibility = ["//visibility:private"],
+)
+
+cc_library(
+    name = "freertos_config",
+    hdrs = [
+        "FreeRTOSConfig.h",
+    ],
+    includes = ["."],
+    deps = [
+        "//sw/device/lib/arch:device",
+    ],
+)
```

Comment:
Is the reason for breaking this config out into a separate build target so that you can build the FreeRTOS kernel as its own lib to include as an OTTF dep? If so, neat! I like this architecture a lot better.

Also, I just realized, that I don't think `FreeRTOSConfig.h` needs to include `.../device.h`. I believe this was left over from a previous setup and I forgot to remove that #include from `FreeRTOSConfig.h`. Sorry about that.

Reply:
Yes, but its also because of the directory-hierarchy separation between this file and the rest of the FreeRTOS kernel.  Writing rules which use files across package boundaries is rather complex and ugly in bazel.

BTW, this is also the reason `portmacro` is its own target in the freertos kernel directory: it's user-supplied and target architecture specific, so it gets to be its own target.